lib/charms/spark.py

```diff
@@ -19,12 +18,13 @@ def __init__(self, dist_config):
         }
         self.verify_resources = utils.verify_resources(*self.resources.values())
 
-    def is_installed(self):
-        return unitdata.kv().get('spark.installed')
 
-    def install(self, force=False):
-        if not force and self.is_installed():
-            return
+    def install(self):
+        '''
+        install Spark and add dependencies in dist-config
+        '''
+        self.dist_config.add_dirs()
+        self.dist_config.add_packages()
```

Comment:
I'm realizing that the `hadoop-client` layer should either act on the options it defines (in which case it would need to expose another state to indicate when that was done), or not define the options at all and leave it up to the client charm.

I'm not sure which would be better / more user-friendly.